### ISSUE TITLE
Transformation of relative pose to absolute pose

### ISSUE THREAD

**Issue by lucleon** [ChiWeiHsiao/DeepVO_PyTorch#30]
In line 94 in test.py, the Rotation Matrix is computed using only one Euler angle:

[ang = eulerAnglesToRotationMatrix([0, answer[-1][0], 0])](url)

I'm wondering, why are theta_x and theta_z  set to zero?

**Comment by fangxu622** [comment]
@ChiWeiHsiao @lucleon  Hi , I have same questions.

**Comment by yangxiaoliyang** [comment]
I have the same problem, and when I evaluate the error of relative pose, the error is very large, but the trajectory fits well. Have you solved it? @fangxu622 @lucleon @ChiWeiHsiao

**Comment by mrdavoodi** [comment]
KITTI is recorded from the trajectory of a car. This movement does not have pitch and roll. The only angle change is yaw.

**Comment by ztysdu** [comment]
> I have the same problem, and when I evaluate the error of relative pose, the error is very large, but the trajectory fits well. Have you solved it? @fangxu622 @lucleon @ChiWeiHsiao

In my opinion,This is due to in test.py,they measure the loss in absolute pose,which is different from training

**Comment by yangxiaoliyang** [comment]
It's worked out. This is the absolute pose. It can be solved by processing the pose data in the data set as relative pose.Hope to help you!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ChiWeiHsiao/DeepVO-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月22日(星期二) 上午9:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ChiWeiHsiao/DeepVO-pytorch] Transformation of relative pose to absolute pose (#30)





  
I have the same problem, and when I evaluate the error of relative pose, the error is very large, but the trajectory fits well. Have you solved it? @fangxu622 @lucleon @ChiWeiHsiao
  
In my opinion,This is due to in test.py,they measure the loss in absolute pose,which is different from training

**Comment by yangxiaoliyang** [comment]
Thank you for your reply！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ChiWeiHsiao/DeepVO-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月22日(星期二) 上午9:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ChiWeiHsiao/DeepVO-pytorch] Transformation of relative pose to absolute pose (#30)





  
I have the same problem, and when I evaluate the error of relative pose, the error is very large, but the trajectory fits well. Have you solved it? @fangxu622 @lucleon @ChiWeiHsiao
  
In my opinion,This is due to in test.py,they measure the loss in absolute pose,which is different from training
